src/calendar/MonthPicker.ts

```diff
@@ -0,0 +1,311 @@
+		}
+		else {
+			onRequestMonthChange && onRequestMonthChange(month - 1);
+		}
+	}
+
+	private _onMonthIncrease() {
+		const {
+			month,
+			year,
+			onRequestMonthChange,
+			onRequestYearChange
+		} = this.properties;
+
+		if (month === 11) {
+			onRequestMonthChange && onRequestMonthChange(0);
+			onRequestYearChange && onRequestYearChange(year + 1);
+		}
+		else {
+			onRequestMonthChange && onRequestMonthChange(month + 1);
+		}
+	}
+
+	private _onPopupKeyDown(event: KeyboardEvent) {
+		// close popup on escape, or if a value is selected with enter/space
+		if (
+			event.which === Keys.Escape ||
+			event.which === Keys.Enter ||
+			event.which === Keys.Space
+		) {
+			this._closePopup();
+		}
+	}
+
+	private _onRadioChange(event: TypedTargetEvent<HTMLInputElement>) {
+		const {
+			onRequestMonthChange
+		} = this.properties;
+		const month = parseInt(event.target.value, 10);
+
+		onRequestMonthChange && onRequestMonthChange(month);
+	}
+
+	private _onYearKeyDown(event: KeyboardEvent) {
+		switch (event.which) {
+			case Keys.Right:
+				this._increaseYear();
+				break;
+			case Keys.Left:
+				this._decreaseYear();
+				break;
+		}
+	}
+
+	private _renderMonthLabel(month: number, year: number) {
+		const { monthNames, renderMonthLabel } = this.properties;
+		return renderMonthLabel ? renderMonthLabel(month, year) : `${monthNames[month].long} ${year}`;
+	}
+
+	private _renderMonthRadios() {
+		const { month, open } = this.properties;
+
+		return this.properties.monthNames.map((monthName, i) => v('label', {
+			key: `${this._idBase}_radios_${i}`,
+			classes: this.classes(css.monthRadio, i === month ? css.monthRadioChecked : null)
+		}, [
+			v('input', {
+				checked: i === month,
+				classes: this.classes(css.monthRadioInput),
+				name: `${this._idBase}_radios`,
+				tabIndex: open ? 0 : -1,
+				type: 'radio',
+				value: i + '',
+				onchange: this._onRadioChange,
+				onmouseup: this._closePopup
+			}),
+			v('abbr', {
+				classes: this.classes(css.monthRadioLabel),
+				title: monthName.long
+			}, [ monthName.short ])
+		]));
+	}
+
+	protected render(): DNode {
+		const {
+			month,
+			year,
+			labelId = `${this._idBase}_label`,
+			labels,
+			open = false
+		} = this.properties;
+
+		return v('div', { classes: this.classes(css.header) }, [
+			// button
+			v('button', {
+				key: 'button',
+				'aria-controls': `${this._idBase}_dialog`,
+				'aria-describedby': labelId,
+				'aria-expanded': String(open),
+				'aria-haspopup': 'true',
+				classes: this.classes(css.monthTrigger),
+				id: `${this._idBase}_button`,
+				onclick: this._onButtonClick
+			}, [
+				v('span', { classes: this.classes().fixed(baseCss.visuallyHidden) }, [ labels.chooseMonth ])
+			]),
+			v('label', {
+				id: labelId,
+				classes: this.classes(css.currentMonthLabel),
+				'aria-live': 'polite',
+				'aria-atomic': 'false'
+			}, [ this._renderMonthLabel(month, year) ]),
+			// previous/next month buttons
```

Comment:
Hah, @tomdye said the same thing. It's a pretty large chunk of VDOM, though. I have trouble finding what I'm looking for at a glance, and I wrote it 😕. Are you sure we shouldn't have comments in there for later?